### PR TITLE
Use MailService to generate Emails

### PR DESCRIPTION
**Pull request by simonseyock**
This is an extension to the v3 branch to be able to override the email generation.

### CODE REVIEW

File: shogun-lib/src/main/java/de/terrestris/shoguncore/service/MailService.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+    /**
+     * Generates a simple email to send the confirmation the user has reset the password, with the new password
+     * @param user
+     * @param password
+     * @param locale
+     * @return SimpleMailMessage
+     */
+    public void sendPasswordResetConfirmedEmailMessage(final User user,
+                                                                          final String password, Locale locale){
```
Comment (dnlkoch) [inline shogun-lib/src/main/java/de/terrestris/shoguncore/service/MailService.java:34]: Please check indentation.

File: shogun-lib/src/main/java/de/terrestris/shoguncore/service/MailService.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+     *
+     * @param appUrl Base URL of the application. Must be provided without a trailing slash. The confirm registration link is
+     *              built from this URL.
+     * @param user The user about to be registered.
+     * @param token The unique token (e.g. UUID).
+     * @return A SimpleMailMessage translated depending on the users' language containing the specified text and a
+     * registration link.
+     */
+    public void sendRegistrationEmailMessage(String appUrl, final User user,
+                                                                final String token, Locale locale) throws MessagingException {
```
Comment (dnlkoch) [inline shogun-lib/src/main/java/de/terrestris/shoguncore/service/MailService.java:67]: Please check indentation.

File: shogun-lib/src/main/java/de/terrestris/shoguncore/service/MailService.java [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+     *
+     * @param appUrl Base URL of the application. Must be provided without a trailing slash. The confirm registration link is
+     *              built from this URL.
+     * @param user The user about to be registered.
+     * @param token The unique token (e.g. UUID).
+     * @return A SimpleMailMessage translated depending on the users' language containing the specified text and a
+     * registration link.
+     */
+    public void sendPasswordResetEmailMessage(String appUrl, final User user,
+                                                                 final String token, Locale locale) throws MessagingException {
```
Comment (dnlkoch) [inline shogun-lib/src/main/java/de/terrestris/shoguncore/service/MailService.java:101]: Please check indentation.

**Review by dnlkoch — APPROVED** [review]

File: shogun-lib/src/main/java/de/terrestris/shoguncore/service/MailService.java (hunk 1 above)
Comment (simonseyock) [reply shogun-lib/src/main/java/de/terrestris/shoguncore/service/MailService.java:34]: copy paste errors :grimacing:

**Review by Unraveler — APPROVED** [review]
Looks good to me

**Pull request merged**